adaptive_memory_v4.0.py

```
@@ -0,0 +1,1769 @@
+                            logger.info(f"Skipping duplicate memory: '{content[:50]}...'")
+                            continue
+
+                    # Format: [Tags: tag1, tag2] Content [Memory Bank: Bank] [Confidence: X.XX]
+                    tags_str = ", ".join(tags) if tags else "none"
+                    confidence = op.get("confidence", 1.0)
+
+                    final_content = f"[Tags: {tags_str}] {content} [Memory Bank: {bank}] [Confidence: {confidence:.2f}]"
+
+                    # Add memory - using Model directly
+                    try:
+                        mem_obj = Memories.insert_new_memory(user_id, final_content)
```

Ruff (0.14.10)

Local variable `mem_obj` is assigned to but never used

Remove assignment to unused variable `mem_obj`

(F841)